src/Compilers/CSharp/Test/Symbol/Compilation/SemanticModelGetSemanticInfoTests.cs

```diff
@@ -5553,6 +5555,124 @@ void M()
             Assert.True(semanticInfo.Symbol.GetSymbol().IsFromCompilation(compilation));
         }
 
+        [WorkItem(59709, "https://github.com/dotnet/roslyn/issues/59709")]
+        [Fact]
+        public void LocalForEach()
+        {
+            string sourceCode = @"
+class C
+{
+    void M()
+    {
+        var a = new int[] { 1, 2, 3 };
+        foreach (var x in a)
+        {
+            var y = /*<bind>*/x/*</bind>*/;
+        }
+    }
+}
+";
+            var compilation = CreateCompilation(sourceCode);
+            var semanticInfo = GetSemanticInfoForTest<IdentifierNameSyntax>(compilation);
+
+            var symbol = (ILocalSymbol)semanticInfo.Symbol;
+            Assert.True(symbol.IsForEach);
+        }
+
+        [WorkItem(59709, "https://github.com/dotnet/roslyn/issues/59709")]
+        [Fact]
+        public void LocalForEachAwait()
+        {
+            string sourceCode = @"
+class C
+{
+    void M()
+    {
+        var a = new int[] { 1, 2, 3 };
+        await foreach (var x in a)
+        {
+            var y = /*<bind>*/x/*</bind>*/;
+        }
+    }
+}
+";
+            var compilation = CreateCompilation(sourceCode);
+            var semanticInfo = GetSemanticInfoForTest<IdentifierNameSyntax>(compilation);
+
+            var symbol = (ILocalSymbol)semanticInfo.Symbol;
+            Assert.True(symbol.IsForEach);
```

Comment:
>IsForEach [](http://example.com/codeflow?start=31&length=9)

I think we need to add a test that consumes VB extension method from Microsoft.CodeAnalysis.VisualBasic.VisualBasicExtensions with an assert of this form ```Assert.True(symbol.IsForEach())```. It used to be a valid code prior to addition of ILocalSymbol.IsForEach property. Is it still a valid code?

Reply:
For this test, we will need to get a local symbol from VB compilation rather than from C# compilation (otherwise the extension method is going to throw). We have a helper to do that - ```CreateVisualBasicCompilation```. There are several tests in this assembly that use it.

Reply:
As already mentioned, `symbol.IsForEach` is still a valid code (although now it calls the property, so it will return `true` also for C# locals whereas previously the extension method returned `true` only for VB locals). Added tests for it: 71bbb2cc22a74c7384e10adef4f2393be42670aa